src/components/Button/Button.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { forwardRef } from 'react'
+import styled from 'styled-components'
+import { space, layout, flexbox, background, variant } from 'styled-system'
+import PropTypes from 'prop-types'
+
+import { Typography } from '../'
+
+const CustomButton = forwardRef(
+  ({ type, variant, children, color, fontSize = 'button', disabled, fontWeight, ...props }, ref) => {
+    const nameColor = color.split('.') // Apenas para mostrar a variação de cores
+    return (
+      <Button
+        type={type}
+        color={color}
+        colorVariant={nameColor}
+        ref={ref}
+        variant={disabled ? 'disabled' : variant}
+        disabled={disabled}
+        {...props}
+      >
+        <Typography
+          color={variant === 'filled' || disabled ? 'white' : color}
+          fontSize={fontSize}
+          fontWeight={fontWeight}
+        >
+          {children}
+        </Typography>
+      </Button>
+    )
+  }
+)
+
+const Button = styled.button`
+  width: 100%;
+  border-width: 2px;
+  border-radius: 4px;
+  border-style: solid;
+  box-sizing: border-box;
+  transition: all 0.5s ease-in-out;
+  &:hover {
+    cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
+  }
+  ${({ color, colorVariant }) =>
+    variant({
+      variants: {
+        filled: {
+          backgroundColor: color,
+          borderColor: color,
+          '&:hover': {
+            backgroundColor: `${colorVariant[0]}.600`,
+            borderColor: `${colorVariant[0]}.600`
+          },
+          '&:active': {
+            backgroundColor: `${colorVariant[0]}.800`,
+            borderColor: `${colorVariant[0]}.800`
+          }
+        },
+        outline: {
+          backgroundColor: 'transparent',
+          borderColor: color,
+          '&:hover': {
+            borderColor: `${colorVariant[0]}.600`
+          },
+          '&:active': {
+            borderColor: `${colorVariant[0]}.800`
+          }
+        },
+        text: {
+          backgroundColor: 'transparent',
+          borderColor: 'transparent'
+        },
+        disabled: {
+          backgroundColor: 'gray.500',
+          borderColor: 'gray.500'
+        }
+      }
+    })}
+`
+Button.defaultProps = {
+  type: 'submit',
+  variant: 'filled',
+  color: 'gray.500'
+}
+
+Button.propTypes = {
+  type: PropTypes.string,
+  color: PropTypes.string,
+  fontString: PropTypes.string,
+  fontWeight: PropTypes.string,
+  variant: PropTypes.string,
+  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
+}
+
+export default CustomButton
```

Comment:
Fala @mathfigue

Dei uma boa revisada no teu código e a funcionalidade do botão ta 100%. Eu fui fazendo pequenas alterações no teu código e etapa a etapa eu acabei abstraindo algumas `props` pra tentar usar o máximo do styled-system possível, e acabei chegando em um componente praticamente 100% styled (só to usando um componente React pra passar a ref do botão adiante e pros PropTypes funcionarem direitinho).
O único ponto onde eu ainda não to confortável é com as cores. Eu vi que tu fez um trabalho legal mas não ficou tão flexível, pq naquela estrutura tu necessariamente precisa utilizar uma cor do theme pra pegar as variações, e como a gente ainda não tem uma estrutura tão boa pra usar essas variações do theme, por hora optei por manter 3 props separadas (`color`, `hoverColor` e `activeColor`) pra gente poder utilizar esse componente o quanto antes, e quando a gente implementar o sistema de paleta de cores mais customizável (alô @dpnunez ) a gente volta aqui e melhora essa estrutura.
Quero uma revisão bem sincera do que tu achou do approach que eu tive e se tu não concordar com algo a gente conversa sobre, fica aqui minha sugestão de código <3

```suggestion
import React, { forwardRef } from 'react'
import styled from 'styled-components'
import { space, border, layout, color, typography, variant } from 'styled-system'
import PropTypes from 'prop-types'
import StyledSystemProptypes from '@styled-system/prop-types'

const Button = forwardRef((props, ref) => <Base ref={ref} {...props} />)

const Base = styled.button`
  cursor: pointer;
  ${color}
  ${space}
  ${border}
  ${layout}
  ${typography}
  ${({ color, hoverColor, activeColor }) =>
    variant({
      variants: {
        filled: {
          backgroundColor: color,
          borderColor: color,
          color: 'white',
          '&:hover': {
            backgroundColor: hoverColor,
            borderColor: hoverColor
          },
          '&:active': {
            backgroundColor: activeColor,
            borderColor: activeColor
          },
          '&:disabled': {
            backgroundColor: 'gray.500',
            borderColor: 'gray.500',
            color: 'white',
            cursor: 'initial'
          }
        },
        outline: {
          backgroundColor: 'transparent',
          borderColor: color,
          '&:hover': {
            borderColor: hoverColor,
            color: hoverColor
          },
          '&:active': {
            borderColor: activeColor,
            color: activeColor
          },
          '&:disabled': {
            borderColor: 'gray.500',
            color: 'gray.500',
            cursor: 'initial'
          }
        },
        text: {
          backgroundColor: 'transparent',
          borderColor: 'transparent',
          '&:hover': {
            color: hoverColor
          },
          '&:active': {
            color: activeColor
          },
          '&:disabled': {
            color: 'gray.500',
            cursor: 'initial'
          }
        }
      }
    })}
`

Button.defaultProps = {
  p: 3,
  borderWidth: '1px',
  borderStyle: 'solid',
  borderRadius: 4,
  fontSize: 'button',
  fontWeight: 'button',
  color: 'blue.main',
  hoverColor: 'blue.600',
  activeColor: 'blue.800'
}

Button.propTypes = {
  ...StyledSystemProptypes.color,
  ...StyledSystemProptypes.space,
  ...StyledSystemProptypes.border,
  ...StyledSystemProptypes.layout,
  ...StyledSystemProptypes.typography,
  color: PropTypes.string,
  hoverColor: PropTypes.string,
  activeColor: PropTypes.string
}

export default Button
```